src/main/java/joe/Joe.java

```diff
@@ -0,0 +1,239 @@
+package joe;
+
+import java.io.IOException;
+import java.io.FileWriter;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+
+public class Joe {
+    private static final String userHome = System.getProperty("user.home");
+    private static final String INTENDATION = "      ";
+    private static final String FILE_PATH = userHome + "/tasks.txt";
+    private static final String SEPARATOR = "_________________________________________________";
+    private static ArrayList<Task> toDoItemArrayList = new ArrayList<Task>();
+    private static final String LOGO = "    (_)           \n"
+            + INTENDATION + "     _  ___   ___ \n"
+            + INTENDATION + "    | |/ _ \\ / _ \\\n"
+            + INTENDATION + "    | | (_) |  __/\n"
+            + INTENDATION + "    | |\\___/ \\___|\n"
+            + INTENDATION + "   _/ |           \n"
+            + INTENDATION + "  |__/            \n";
+
+    public static void main(String[] args) {
+
+        File file = new File(FILE_PATH);
+        if (file.exists()) {
+            try {
+                readFromFile(FILE_PATH);
+            } catch (IOException ioException) {
+                System.out.println("Something went wrong during loading tasks.");
+            }
+        } else {
+            try {
+                file.createNewFile();
+            } catch (IOException ioException) {
+                System.out.println("Creating a file for storing the tasks was unsuccessful.");
+            }
+        }
+
+        printGreeting();
+        chatWithJoe();
+        printFarewell();
+
+        try {
+            writeToFile(FILE_PATH);
+        } catch (IOException e) {
+            System.out.println("File not found!");
+        }
+    }
+
+    public static void chatWithJoe() {
+        //get and echo user input
+        Scanner in = new Scanner(System.in);
+        String input = "";
+        while (!input.equals("bye")) {
+            input = in.nextLine();
+            switch (input) {
+            case "bye":
+                break;
+            case "list":
+                printList();
+                break;
+            default:
+                chatAboutTasks(input);
+            }
+        }
+    }
+
+
+    // helper methods
+
+    public static void chatAboutTasks(String input) {
```

Comment:
This method is pretty long. You can consider extracting each case into a separate method. It will be better to keep methods short.